addons/mail/models/discuss/ir_websocket.py

```diff
@@ -21,6 +21,45 @@ def _get_im_status(self, data):
             )]
         return im_status
 
+    def _get_missed_presences_identity_domains(self, presence_channels):
+        identity_domain = super()._get_missed_presences_identity_domains(presence_channels)
+        if guest_ids := [
+            g.id for g, _ in presence_channels if isinstance(g, self.pool["mail.guest"])
+        ]:
+            identity_domain.append([("guest_id", "in", guest_ids)])
+        return identity_domain
+
+    @add_guest_to_context
+    def _build_presence_channel_list(self, presences):
+        channels = super()._build_presence_channel_list(presences)
+        self_discuss_channel_ids = []
```

Comment:
prefer keeping recordsets as long as possible, only using `.ids` at the last moment